### PR TITLE
Raspi pipeline

### PR DESCRIPTION
**Pull request by YoDaMa**
# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `master` branch. 
  - [ ] I have merged the latest `master` branch prior to submission and re-merged as needed after I took any feedback.
  - [ ] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: iothub_client/CMakeLists.txt [hunk 1]
```diff
@@ -10,6 +10,7 @@ endif()
 
 compileAsC99()
```
Comment (jebrando) [inline iothub_client/CMakeLists.txt]: remove this file.

File: jenkins/raspberrypi/run_tests.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+chmod -R 777 $PWD/drop
```
Comment (jebrando) [inline jenkins/raspberrypi/run_tests.sh]: I'm pretty sure you need a set -e to return errors out of this file to validate failures.

File: jenkins/raspberrypi/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+# get ready for ssh keys
+RUN mkdir -p /root/.ssh && chmod 700 /root/.ssh && touch /root/.ssh/authorized_keys && chmod 600 /root/.ssh/authorized_keys
+
+# make a key that we can use to connect to SSH and authorize it.  
+# people who want to use SSH can use docker cp to get the key out of the container
+RUN ssh-keygen -b 2048 -t rsa -f /root/.ssh/remote-debug -q -N ""
+RUN cat /root/.ssh/remote-debug.pub > /root/.ssh/authorized_keys
+
+#RUN mkdir RPiTools
```
Comment (bikamani) [inline jenkins/raspberrypi/Dockerfile]: Do we still need this ?

File: jenkins/raspberrypi/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+# Finally a sanity check to make sure the files are there
+RUN ls -al ${TOOLCHAIN_PREFIX}/lib
+RUN ls -al ${TOOLCHAIN_PREFIX}/include
+
+RUN ls -la ./
+
+# Go to project root
+WORKDIR /
+
```
Comment (bikamani) [inline jenkins/raspberrypi/Dockerfile]: Nit: remove extra new lines

**Comment by massand** [comment]
@YoDaMa - which e2e tests have we managed to successfully run so far? Can they run back to back? I believe mqtt_e2e was the one you ran for this, which is sufficient for nightly build at the moment. However, it would be good to know what the rest of the list is - as part of the PR comments.

**Pull request merged**